great_expectations_provider/operators/great_expectations.py

```diff
@@ -78,24 +64,22 @@ class GreatExpectationsOperator(BaseOperator):
     ui_color = "#AFEEEE"
     ui_fgcolor = "#000000"
     template_fields = (
-        "checkpoint_name",
-        "batch_kwargs",
-        "assets_to_validate",
         "data_context_root_dir",
+        "data_context_config",
```

Comment:
@talagluck The `jinja` templating won't work the way we want on the `data_context_config` and `checkpoint_config` variables, unless the `DataContextConfig` and `CheckpointConfig` classes themselves have a `templated_fields` class variable, which I doubt (nor should they, IMO, as that would leak an Airflow specific convention into the core GE code base).

If the idea is to allow for run-time `jinja` templating against these two config objects, I'm pretty sure we'll have to do one of two things:
1. Add two `dict` parameters/variables (eg, `data_context_config_dict` and `checkpoint_config_dict`), mark those as templated, allow the caller to pass them into the operator constructor, and then create the `DataContextConfig` and `CheckpointConfig` objects themselves with those `dict` objects inside `execute()`.  This is essentially _almost_ already happening in the example DAG, in that the config objects are just wrapping hard coded `dict` objects.
2. Add the `create_data_context()` function back in (and add something similar for the checkpoint), allowing the caller to override it and dynamically create the context at run-time.

Reply:
Thanks @jeffkpayne - this is great, and not something that I had known. As far as I'm concerned, I would be OK removing these from the `template_fields`, as the most important field to template for me is `checkpoint_kwargs`. As a user, do you feel that that is sufficient, or do you think it would be important to template these as well?

Reply:
@talagluck It's not clear to me if those should be templated.  Essentially, anything that is potentially environment specific should be templated.

For instance, for AF operators that access GCP resources (BigQuery, GCS, etc), the `GCLOUD_PROJECT` parameter/variable should be templated so it can be resolved at DAG execution time by pulling the value in from the metadata DB, without incurring the cost of constantly hitting the DB when AF scans the DAG folder.

So, if everything that's environment specific (database connections, GCP projects, etc) can be kept out of the `data_context_config` and `checkpoint_config` objects and put into `checkpoint_kwargs`, then it should be fine to not template those objects.

Where do environment specific config values typically go?  Would `evaluation_parameters` or `runtime_configuration` be used for this type of thing?

Reply:
Interesting - for most GE workflows, we rely on credentials that are either explicitly declared in the DataContext, available in environment variables, or contained in a secrets store. Do you think these would be sufficient? If not, then I think the way to do would be to add new methods to the operator that update the Config objects.

Reply:
Right, that's kind of what I thought. With Airflow, typically credentials for db connections etc are stored specifically as Airflow "connections" in the metadata db. That said, if the underlying Python library connecting to said resource can fall back to environment variables or local file based configuration, then maybe that can be the recommended approach. This is actually what we rely on for AWS and GCP in some cases.